src/Command/RecipesCommand.php

```diff
@@ -250,4 +267,56 @@ private function writeTreeLine($line)
 
         $io->write($line);
     }
+
+    /**
+     * Attempts to find the original git sha when the recipe was installed.
+     */
+    private function findRecipeShaFromTreeRef(string $package, string $repo, string $branch, string $version, string $lockRef)
+    {
+        // only supports public repository placement
+        if (0 !== strpos($repo, 'github.com')) {
+            return null;
+        }
+
+        $parts = explode('/', $repo);
+        if (3 !== \count($parts)) {
+            return null;
+        }
+
+        $recipePath = sprintf('%s/%s', $package, $version);
+        $commitsData = $this->requestGitHubApi(sprintf(
```

Comment:
We also have to try/catch if the api isn't available and I think after that the command should work even offline